File: server/graph/model/roadmap.go
package model

import "time"

type Roadmap struct {
	ID          int       `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	CreatorID   int       `json:"creator"`
	Checkpoints []int     `json:"checkpoints"`
	Followers   []int     `json:"followers"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

File: web/graphql/fragments/roadmap.ts
import { gql } from "@apollo/client";

export const ROADMAP_INFO_FIELDS = gql`
  fragment RoadmapInfoFields on Roadmap {
    id
    title
    description
    createdAt
    updatedAt
    creator {
      id
      username
    }
    followers {
      id
    }
    checkpoints {
      id
    }
  }
`;
